Dev

haechangchoi opened the pull request:
## 🔧 어떤 기능인가요?

> 회원 CRUD 기능을 수정 및 개선하는 작업입니다.

## #️⃣ 연관된 이슈

> 관련 이슈 없음

## 💡 리뷰어에게 하고 싶은 말

- [x] `findByMemberId` -> `findById`로 변경.  
- [x] Member 엔티티 경로가 달라 코드 부분을 수정.  
- [x] ModelMapper를 추가하여 사용.  
- [x] soft delete 관련 고도화는 현재 보류 상태이며, 추후 고려 예정.  


<!-- 코드에 나타나지 않는 설계 및 의도, 중점적으로 봐줬으면 하는 점, 특이한 변경사항 등 -->

## 🙏 아래 내용이 모두 충족 되었는지 확인해주세요 🙏

- [x] PR 이전 `dev` 브랜치 병합 하셨나요?  
- [x] PR 이전 빌드 테스트 정상적으로 성공했나요?  
- [x] PR 상세내용이 충분히 기재 되었나요?  
- [x] PR 리뷰어, 할당자, 라벨, 프로젝트 확인

haechangchoi committed:
Merge pull request #1 from pickone-band/dev

initial setting

haechangchoi committed:
feat: 회원 엔티티

기본적인 회원 엔티티

haechangchoi committed:
feat: 회원 CRUD

Member, MemberState, Term, MemberTerm, Profile, SocialAccount

haechangchoi committed:
Merge branch 'dev' of https://github.com/pickone-band/PickOne-be into dev

haechangchoi committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into dev

haechangchoi committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/com/PickOne/domain/user/repository/MemberRepository.java

haechangchoi committed:
feat: 회원 CRUD Service Layer

- findByMemberId -> findById로 변경
- Member 엔티티 경로가 달라 상대방 코드 부분 수정
- ModelMapper 추가 및 사용
- soft delete 관련 코드 고도화는 추후 고려

Member, MemberState, Term, MemberTerm, Profile, SocialAccount

kendrickhama commented on src/main/java/com/PickOne/domain/user/service/MemberService.java:
```diff
@@ -0,0 +1,90 @@
+    private final ModelMapper modelMapper;
+    private final PasswordEncoder passwordEncoder;
+
+        /** 회원가입 */
+        public MemberResponseDto createMember(MemberCreateDto dto) {
+            validateDuplicate(dto);
+
+            Member member = modelMapper.map(dto, Member.class);
+            member.setRole(Role.USER);
+            member.setPassword(passwordEncoder.encode(dto.getPassword()));
+
+            Member saved = memberRepository.save(member);
+            // DTO 변환 시 id 수동 설정
+            MemberResponseDto responseDto = modelMapper.map(saved, MemberResponseDto.class);
+            responseDto.setId(saved.getId()); // id 직접 매핑
+
+            return responseDto;
+        }
```
- [ ] 회원가입하고 dto를 반환하는 이유가 뭔가요?

haechangchoi committed:
feat: Swagger 설정 추가

- build.gradle에 springdoc-openapi 의존성 추가
- SwaggerConfig.java 파일 생성 및 Swagger 설정 추가
- SecurityConfig.java에 Swagger 관련 보안 설정 적용
